nibabel/freesurfer/io.py

```diff
@@ -44,21 +45,64 @@ def _fread3_many(fobj, n):
     return (b1 << 16) + (b2 << 8) + b3
 
 
-def read_geometry(filepath):
+def _read_volume_info(extra):
+    volume_info = OrderedDict()
+
+    if extra is None:
+        return volume_info
+
+    if extra[:4] != b'\x00\x00\x00\x14':
+        warnings.warn("Unknown extension code.")
+    else:
+        try:
+            for line in extra[4:].split(b'\n'):
+                if len(line) == 0:
+                    continue
+                key, val = map(bytes.strip, line.split(b'=', 1))
+                key = key.decode('utf-8')
+                if key in ('voxelsize', 'xras', 'yras', 'zras', 'cras'):
+                    val = np.fromstring(val, sep=' ')
+                    val = val.astype(np.float)
+                elif key == 'volume':
+                    val = np.fromstring(val, sep=' ', dtype=np.uint)
+                    val = val.astype(np.int)
+                volume_info[key] = val
+        except ValueError:
+            raise ValueError("Error parsing volume info")
```

Comment:
This is a lot of lines to be put in `try`. Which line is expected to fail? If it's only one (hopefully) then it's better to only have that line in the `try` clause.

If you're trying to catch any possible error on the other hand, then `except Exception` is more appropriate, and it would be good to do `"Error parsing volume info: '%s'" % err`